librz/util/str.c

```diff
@@ -2474,7 +2474,7 @@ RZ_API size_t rz_str_len_utf8_ansi(const char *str) {
 			i += chlen - 1;
 		} else if ((ch & 0xc0) != 0x80) { // utf8
 			len++;
-			if (rz_str_char_fullwidth(str + i, 4)) {
+			if (rz_str_char_fullwidth(str + i, chlen)) {
```

Comment:
Probably better to replace `chlen` with `strlen(str + i)` since if `chlen` is always 1, the test is surely always false, but the `strlen` seems expensive, there are no tests and I'm not that familiar with fullwidth chars anyway.

Reply:
This whole function looks a bit wrong to me, to be honest. It fixed the crash and didn't break the test. But in the long run it's better to write a test, and likely rewrite it.

Reply:
That's probably because the OOB peek-a-boo didn't change the falseness of the test.

Reply:
That's probably because the OOB peek-a-boo didn't change the falseness of the `if` condition.